Compile errors with gcc/4.7.1

PIC does NOT compile with gcc 4.7.1 (tested on machine joker).
Compile error is:
`[ 83%] Built target cuda_memtest
/home.local/others/fschmitt/pic/picongpu/src/libPMacc/include/mappings/kernel/ExchangeMapping.hpp: In instantiation of ‘PMacc::DataSpace<DIM> PMacc::ExchangeMapping<areaType, baseClass<DIM, SuperCellSize_> >::getGridDim() [with unsigned int areaType = 4u; baseClass = PMacc::MappingDescription; unsigned int DIM = 3u; SuperCellSize_ = PMacc::TVec<8u, 8u, 4u>]’:
/home.local/others/fschmitt/pic/picongpu/src/picongpu/include/fields/FieldJ.tpp:120:31:   required from here
/home.local/others/fschmitt/pic/picongpu/src/libPMacc/include/mappings/kernel/ExchangeMapping.hpp:97:87: error: ‘reduce’ was not declared in this scope, and no declarations were found by argument-dependent lookup at the point of instantiation [-fpermissive]
/home.local/others/fschmitt/pic/picongpu/src/libPMacc/include/mappings/kernel/ExchangeMapping.hpp:97:87: note: declarations in dependent base ‘PMacc::CudaGridDimRestrictions<3u>’ are not found by unqualified lookup
/home.local/others/fschmitt/pic/picongpu/src/libPMacc/include/mappings/kernel/ExchangeMapping.hpp:97:87: note: use ‘this->reduce’ instead
/home.local/others/fschmitt/pic/picongpu/src/libPMacc/include/mappings/kernel/AreaMapping.hpp: In instantiation of ‘PMacc::DataSpace<DIM> PMacc::AreaMapping<areaType, baseClass<DIM, SuperCellSize_> >::getGridDim() [with unsigned int areaType = 3u; baseClass = PMacc::MappingDescription; unsigned int DIM = 3u; SuperCellSize_ = PMacc::TVec<8u, 8u, 4u>]’:
/home.local/others/fschmitt/pic/picongpu/src/picongpu/include/plugins/SumCurrents.hpp:190:246:   required from here
/home.local/others/fschmitt/pic/picongpu/src/libPMacc/include/mappings/kernel/AreaMapping.hpp:72:98: error: ‘reduce’ was not declared in this scope, and no declarations were found by argument-dependent lookup at the point of instantiation [-fpermissive]
/home.local/others/fschmitt/pic/picongpu/src/libPMacc/include/mappings/kernel/AreaMapping.hpp:72:98: note: declarations in dependent base ‘PMacc::CudaGridDimRestrictions<3u>’ are not found by unqualified lookup
/home.local/others/fschmitt/pic/picongpu/src/libPMacc/include/mappings/kernel/AreaMapping.hpp:72:98: note: use ‘this->reduce’ instead
/home.local/others/fschmitt/pic/picongpu/src/libPMacc/include/mappings/kernel/StrideMapping.hpp: In instantiation of ‘PMacc::DataSpace<DIM> PMacc::StrideMapping<areaType, stride, baseClass<DIM, SuperCellSize_> >::getGridDim() [with unsigned int areaType = 3u; unsigned int stride = 3u; baseClass = PMacc::MappingDescription; unsigned int DIM = 3u; SuperCellSize_ = PMacc::TVec<8u, 8u, 4u>]’:
/home.local/others/fschmitt/pic/picongpu/src/picongpu/include/fields/FieldJ.tpp:212:125:   required from ‘void picongpu::FieldJ::computeCurrent(ParticlesClass&, uint32_t) [with unsigned int AREA = 3u; ParticlesClass = picongpu::Particles<boost::mpl::vector2<picongpu::ElectronMethods<>, picongpu::ParticlesData<> > >; uint32_t = unsigned int]’
/home.local/others/fschmitt/pic/picongpu/src/picongpu/include/simulationControl/MySimulation.hpp:372:131:   required from here
/home.local/others/fschmitt/pic/picongpu/src/libPMacc/include/mappings/kernel/StrideMapping.hpp:75:114: error: ‘reduce’ was not declared in this scope, and no declarations were found by argument-dependent lookup at the point of instantiation [-fpermissive]
/home.local/others/fschmitt/pic/picongpu/src/libPMacc/include/mappings/kernel/StrideMapping.hpp:75:114: note: declarations in dependent base ‘PMacc::CudaGridDimRestrictions<3u>’ are not found by unqualified lookup
/home.local/others/fschmitt/pic/picongpu/src/libPMacc/include/mappings/kernel/StrideMapping.hpp:75:114: note: use ‘this->reduce’ instead`

--- a/src/libPMacc/include/mappings/kernel/AreaMapping.hpp
+++ b/src/libPMacc/include/mappings/kernel/AreaMapping.hpp
@@ -69,7 +69,7 @@ namespace PMacc
          */
         HINLINE DataSpace<DIM> getGridDim()
         {
-            return reduce(AreaMappingMethods<areaType, DIM>::getGridDim(*this,
+            return this->reduce(AreaMappingMethods<areaType, DIM>::getGridDim(*this,
                                                                         this->getGridSuperCells()));
         }
 

--- a/src/libPMacc/include/mappings/kernel/ExchangeMapping.hpp
+++ b/src/libPMacc/include/mappings/kernel/ExchangeMapping.hpp
@@ -94,7 +94,7 @@ namespace PMacc
          */
         HINLINE DataSpace<DIM> getGridDim()
         {
-            return reduce(ExchangeMappingMethods<areaType, DIM>::getGridDim(*this, exchangeType));
+            return this->reduce(ExchangeMappingMethods<areaType, DIM>::getGridDim(*this, exchangeType));
         }
 
         /**

--- a/src/libPMacc/include/mappings/kernel/StrideMapping.hpp
+++ b/src/libPMacc/include/mappings/kernel/StrideMapping.hpp
@@ -72,7 +72,7 @@ public:
      */
     HINLINE DataSpace<DIM> getGridDim()
     {
-        return reduce((StrideMappingMethods<areaType, DIM>::getGridDim(*this) - offset + Stride - 1) / Stride);
+        return this->reduce((StrideMappingMethods<areaType, DIM>::getGridDim(*this) - offset + Stride - 1) / Stride);
     }
 
     /**

--- a/src/libPMacc/include/memory/buffers/HostBufferIntern.hpp
+++ b/src/libPMacc/include/memory/buffers/HostBufferIntern.hpp
@@ -104,7 +104,7 @@ public:
     void reset(bool preserveData = true)
     {
         __startOperation(ITask::TASK_HOST);
-        setCurrentSize(this->getDataSpace().getElementCount());
+        this->setCurrentSize(this->getDataSpace().getElementCount());
         if (!preserveData)
             memset(pointer, 0, this->getDataSpace().getElementCount() * sizeof (TYPE));
     }

